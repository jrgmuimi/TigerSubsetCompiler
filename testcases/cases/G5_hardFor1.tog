let
    var loop_ctrl : int
    var counter : int
in
    counter := 0;
    for loop_ctrl:=0 to 100 do          // expect to see 105 iterations
        printstring("loop 1 \n");       // all outer loops only exec once
        counter := counter + 1;         // inner loop exec 100 times
        for loop_ctrl:=0 to 100 do      // when inner loop ends, all outer loop ends too
            printstring("loop 2 \n");
            counter := counter + 1;
            for loop_ctrl:=0 to 100 do
                printstring("loop 3 \n");
                counter := counter + 1;
                for loop_ctrl:=0 to 100 do
                    printstring("loop 4 \n");
                    counter := counter + 1;
                    for loop_ctrl:=0 to 100 do
                        counter := counter + 1;
                        printstring("loop 5 \n")
                    end
                end
            end
        end
    end;
    printstring("\n END\n");
    printint(counter)
end