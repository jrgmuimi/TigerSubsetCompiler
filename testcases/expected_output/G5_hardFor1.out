MARS 4.5  Copyright 2003-2014 Pete Sanderson and Kenneth Vollmar

loop 1 
loop 2 
loop 3 
loop 4 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 
loop 5 

 END
105
